.github/workflows/ci.yml

```
@@ -129,8 +129,94 @@ jobs:
+        # The rule is not "these files may never grow". Recording a PRE-EXISTING
+        # finding in a migration this change does not touch is legitimate
+        # bookkeeping, and the sha256 pin makes it honest — the row dies the
+        # moment that file is edited. What is never legitimate is one change
+        # both introducing a file and exempting it. So: an added row may not
+        # name a migration this same change adds or modifies.
+        #
+        # That closes the attack exactly. Landing the unsafe migration first and
+        # exempting it later does not work either: the row must carry the file's
+        # hash, and the changed-only scan rejects the migration on the PR that
+        # introduces it, before any hash exists to record.
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2016:info:14:13: Expressions don't expand in single quotes, use double quotes for that

(shellcheck)